api.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * High-level API entrypoint
+ */
+
+use Glpi\Api\HL\Controller\AbstractController;
+use Glpi\Api\HL\Router;
+use Glpi\Application\ErrorHandler;
+use Glpi\Cache\CacheManager;
+use Glpi\Http\JSONResponse;
+use Glpi\Http\Request;
+use Glpi\Http\Response;
+
+define('GLPI_ROOT', __DIR__);
+define('DO_NOT_CHECK_HTTP_REFERER', 1);
+ini_set('session.use_cookies', 0);
+
+include_once(GLPI_ROOT . '/inc/based_config.php');
+
+// Init loggers
+$GLPI = new GLPI();
+$GLPI->initLogger();
+$GLPI->initErrorHandler();
+
+//init cache
+$cache_manager = new CacheManager();
+$GLPI_CACHE = $cache_manager->getCoreCacheInstance();
+
+$method = $_SERVER['REQUEST_METHOD'];
+$relative_uri = $_SERVER['PATH_INFO'];
+// Ensure uri starts with slash but does not end with a slash
+$relative_uri = '/' . trim($relative_uri, '/');
+
+// If the relative URI starts with /v1/ or is /v1 then we are dealing with a legacy API request
+if (preg_match('/^\/v1(\/|$)/', $relative_uri)) {
+    // Include the legacy API entrypoint and then die
+    $api = new Glpi\Api\APIRest();
+    $api->call();
+    die();
+}
+include_once(GLPI_ROOT . '/inc/includes.php');
+
+$supported_versions = Router::getAPIVersions(true);
+if (preg_match('/^\/v\d+(\/|$)/', $relative_uri)) {
+    // A specific API version has been requested
+    //TODO Plan handling endpoints with specific versions
+    // For now, just remove the version prefix from the URI
+    $relative_uri = preg_replace('/^\/v\d+(\/|$)/', '/', $relative_uri);
+}
+
+$body = file_get_contents('php://input') ?? null;
+
+$headers = getallheaders() ?? [];
```

Comment:
`getallheaders()` is specific to `Apache`. The `ralouphie/getallheaders` should probably be added to GLPI dependencies to handle installation that are not using `Apache`.

Reply:
That package seems to be present already for me.
It is a dependency of guzzlehttp/psr7.

Reply:
It should be added manually to our dependencies, to ensure that we still have it if `guzzlehttp/psr7` decided to remove it from its dependencies.